filament/backend/src/opengl/OpenGLContext.h

```diff
@@ -634,6 +647,27 @@ void OpenGLContext::depthFunc(GLenum func) noexcept {
     });
 }
 
+void OpenGLContext::stencilFunc(GLenum func, GLint ref, GLuint mask) noexcept {
+    // WARNING: don't call this without updating mRasterState
+    update_state(state.raster.stencilFunc, {func, ref, mask}, [&]() {
+        glStencilFunc(func, ref, mask);
+    });
+}
+
+void OpenGLContext::stencilOp(GLenum op) noexcept {
+    // WARNING: don't call this without updating mRasterState
+    update_state(state.raster.stencilOp, op, [&]() {
+        glStencilOp(GL_KEEP, GL_KEEP, op);
```

Comment:
should we be able to set sfail and dpfail too?

Reply:
Yeah I was wondering about that too. I think we might as well. I was concerned it would add additional fields to `RasterState` but I suppose if we go with 64 bits now we have plenty.